deepchecks/tabular/checks/integrity/outlier_detection.py

```diff
@@ -0,0 +1,149 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Outlier detection functions."""
+from typing import Union, List
+
+import numpy as np
+from PyNomaly import loop
+from sklearn.neighbors import NearestNeighbors
+
+from deepchecks.core import CheckResult, ConditionResult, ConditionCategory
+from deepchecks.tabular import Context, SingleDatasetCheck
+from deepchecks.utils import gower_distance
+from deepchecks.utils.dataframes import select_from_dataframe
+from deepchecks.utils.typing import Hashable
+
+__all__ = ['OutlierDetection']
+
+
+class OutlierDetection(SingleDatasetCheck):
+    """Detects outliers in a dataset using the LoOP algorithm.
```

Comment:
Below this "summery line" there should be a more elaborate description explaining that this is "Multivariate" outlier detection (considers multiple features at once) and a bit about the LoOP algo (also link to paper).